addons/payment_razorpay_oauth/controllers/onboarding.py

```diff
@@ -53,6 +53,9 @@ def razorpay_return_from_authorization(self, **data):
             'view_type': 'form',
         }
         redirect_url = f'/web#{urlencode(url_params)}'  # TODO: change to /odoo in saas-17.2!
+        # Check if the authorization code is missing then redirect to provider form.
+        if not authorization_code or authorization_code == "None":
```

Comment:
Let's not pass the `authorization_code` at all if it's `None` on the proxy, so that it's `None` here too, instead of `"None"`.